### PR TITLE
Design doc for model metadata storage

### LINKED ISSUE #2085: Save model meta to OSS when training on PAI platform
**Is your feature request related to a problem? Please describe.**
Current training process on PAI will not save metadata on OSS, so we can't use the `SHOW TRAIN` statement.

**Describe the solution you'd like**
Save metadata with the model like we do on other platforms

**Describe alternatives (optional)**

**Additional Notes**

### PR DESCRIPTION
**Pull request by lhw362950217**
Fixes #2085

### CODE REVIEW

File: doc/design/model_meta_storage.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Some time ago, we have implemented the `SHOW TRAIN` statement, which shows the metadata to user. Recently, we are developing the Model Zoo.  When releasing a trained model to Model Zoo, we are requested to send the metadata along the the model data.  Both features require us to enrich the metadata, unify the metadata storage and make it easy to use.
+
+## The Design
+First, we do not save the model metadata from step go code any more.  Because the real training work may be remote to the image.  We move the saving work to the python code which is doing the real training.  A file named `model_meta.json` is dedicated to store the metadata.  It is written with the following fields.  Basically, we can serialize all fields in `Train ir` to the file.  Additionally, the evaluation result will be stored if it exists.
+```json
+{
+  "OriginalSQL": "SELECT * from train to TRAIN ...",
+  "Estimator": "DNNClassifier",
+  "Attributes":{"model.hidden_units":[10,10],"model.n_classes":3,"train.batch_size":1},
+  "Features":{"feature_columns":[{"FieldDesc":{"name":"sepal_length","dtype":1,"delimiter":"","shape":[1],"is_sparse":false,"vocabulary":null,"MaxID":0}}]},
```
Comment (typhoonzero) [inline doc/design/model_meta_storage.md]: Should save two types of column information: `derived_field_descs` and `feature_columns`.

- `derived_field_descs` is the current `FieldDesc ` map
- `feature_columns` stores the transformation method from raw data to Tensor, e.g. `NumericColumn`

Comment (lhw362950217) [reply doc/design/model_meta_storage.md]: These two types of information are included in the `train ir`, and they will be serialized to the json.

File: doc/design/model_meta_storage.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Model Meta Storage
+
+This article describe the data structure of the metadata for models trained by SQLFlow.  It also describe how we save and load the metadata to/from kinds of `sqlfs`.
```
Comment (typhoonzero) [inline doc/design/model_meta_storage.md]: Define what is metadata beforehand.

File: doc/design/model_meta_storage.md (hunk 1 above)
Comment (typhoonzero) [reply doc/design/model_meta_storage.md]: Save two fields here: `FieldDescs` and `FeatureColumns` they are different.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**